### PR TITLE
feat(modal,block,loader): deprecate disablePadding , noPadding and add support for custom CSS variables

### PR DESCRIPTION
**Pull request by anveshmekala**
**Related Issue:** #3860

This PR will deprecate `noPadding`, `disablePadding` props across the repo and replace them with CSS props instead. List of CSS props added to replace props

- modal : `--calcite-modal-padding`
- loader : `--calcite-loader-padding`
- block : `--calcite-modal-padding`

Moreover, public CSS props ` --calcite-modal-padding-large`,`--calcite-modal-title-text` &`--calcite-modal-content-text` of `modal` are deprecated.

### CODE REVIEW

File: src/components/block/block.tsx [hunk 1]
```diff
@@ -107,7 +107,11 @@ export class Block implements ConditionalSlotComponent, InteractiveComponent {
   /**Block description */
   @Prop() description: string;
 
-  /** When true, removes padding for the slotted content */
+  /**
+   * When true, removes padding for the slotted content
+   *
+   * @deprecated Use `--calcite-block-padding`  CSS variable instead.
```
Comment (driskull) [inline src/components/block/block.tsx:113]: Looks like an extra space before `  CSS`.

@geospatialem do we refer to CSS variables as `variables` or `custom properties`? 

Custom properties (--*): CSS variables

File: src/components/loader/loader.scss [hunk 2]
```diff
@@ -13,14 +13,15 @@ $loader-scale: 54;
...
-  @apply relative mx-auto hidden items-center justify-center py-16 opacity-100;
+  @apply relative mx-auto hidden items-center justify-center opacity-100;
   min-block-size: var(--calcite-loader-size);
   font-size: var(--calcite-loader-font-size);
   stroke: var(--calcite-ui-brand);
   stroke-width: $stroke-width;
   fill: none;
   transform: scale(1, 1);
   animation: loader-color-shift 6s alternate-reverse infinite linear;
+  padding-block: var(--calcite-loader-padding, 4rem);
```
Comment (driskull) [inline src/components/loader/loader.scss:24]: why the 4rem here?

File: src/components/modal/modal.tsx [hunk 3]
```diff
@@ -115,7 +115,11 @@ export class Modal implements ConditionalSlotComponent, OpenCloseComponent {
   /** Sets the background color of the component's content. */
   @Prop({ reflect: true }) backgroundColor: ModalBackgroundColor = "white";
 
-  /** When true, disables spacing to the content area slot. */
+  /**
+   * When true, disables spacing to the content area slot.
+   *
+   * @deprecated  Use `--calcite-modal-padding`  CSS variable instead.
```
Comment (driskull) [inline src/components/modal/modal.tsx:121]: spacing issue here too.

**Review by driskull — COMMENTED** [review]
comments :)

File: src/components/block/block.tsx (hunk 1 above)
Comment (geospatialem) [reply src/components/block/block.tsx:113]: We'd prefer `variables` for consistency, also perhaps better to keep separated from our custom element properties.

File: src/components/loader/loader.scss (hunk 2 above)
Comment (anveshmekala) [reply src/components/loader/loader.scss:24]: replacing py-16, spacing.16 would be 4rem in tailwind.

Comment (driskull) [reply src/components/loader/loader.scss:24]: Can you reference the spacing instead of 4rem? `theme("spacing.16")`?

Comment (driskull) [reply src/components/loader/loader.scss]: The problem with 4rem is that if we change the spacing down the road it won't get updated here.

Comment (anveshmekala) [reply src/components/loader/loader.scss]: agreed. better to refer tailwind utils.

File: src/components/block/block.scss [hunk 4]
```diff
@@ -146,7 +135,8 @@ calcite-handle {
 }
 
 .content--spaced {
-  @apply px-3 py-2;
+  padding-block: var(--calcite-block-padding, 0.5rem);
```
Comment (driskull) [inline src/components/block/block.scss:138]: These values need to be referenced from tailwind as well.

File: src/components/block/block.stories.ts [hunk 5]
```diff
@@ -175,7 +175,7 @@ export const disabled = (): string => html`<calcite-block heading="heading" summ
   </calcite-block-section>
 </calcite-block>`;
 
-export const noPadding = (): string => html` <calcite-panel heading="Properties">
+export const disablePadding = (): string => html` <calcite-panel heading="Properties">
```
Comment (driskull) [inline src/components/block/block.stories.ts:178]: nitpick: paddingDisabled

**Review by driskull — CHANGES_REQUESTED** [review]
Looks good just needs those two vars updated to use tailwind values

**Review by driskull — APPROVED** [review]
👍 🎉 🌮 w00t

**Pull request merged**